goldigen/generator.go

```diff
@@ -5,13 +5,14 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
-	"os"
 	"sort"
 	"strings"
 
 	"gopkg.in/yaml.v2"
 )
 
+var ioutilReadAll = ioutil.ReadAll
```

Comment:
Did you consider passing a mocked reader instead of mocking `ioutil.ReadAll`?

Something like this:

```go
It("should return error when cannot read from input", func() {
    r := new(errorReader)
    Expect(gen.Generate(r, output)).To(MatchError("could not parse type definition: error from errorReader"))
})

type errorReader struct{}

func (errorReader) Read(p []byte) (n int, err error) {
	return 0, errors.New("error from errorReader")
}

```

Reply:
Done injecting the error reader.